Fix zombies vehicle spawning issue

Berthalamew opened the pull request:
- change classification of netgame items to a powerup if the classification is already a vehicle or if it's using a vehicle collection
- remove some unused and duplicated code
- fixed issue with callbacks to GameLoop on dedicated servers

Berthalamew committed:
remove dead code

- remove unused code from RunLoop and Tweeks
- remove empty interface file (not sure why this was created)

Berthalamew committed:
fixup for Infection gametype

- implement scenario module and function to get the global scenario definition
- make the removal of unwanted items happen on both the host and client
- change the category of an netgame item to a powerup instead of keeping it as a vehicle so they don't spawn default vehicles in zombies

Berthalamew committed:
misc edits

- remove function definition for toggleKillVolumes

Berthalamew committed:
change the classification if the classification is already a vehicle

Kantanomo commented on xlive/H2MOD/Modules/MainLoopPatches/RunLoop/RunLoop.cpp:
There is no reason to remove these, kill volume stuff will be used in the future.

Berthalamew replied:
I'll move this into another module and clean up the functions

Berthalamew committed:
re-add kill volume functionality

rewrite functions and add server offsets

Berthalamew committed:
fix event callbacks for GameLoop on dedis

- implement a hook for main_loop_body on dedicated servers so we can call GameLoopEventExecute
- make it so infection only ends when there's more than 1 zombie rather than 1

nukeulater commented on xlive/H2MOD/Modules/MainLoopPatches/RunLoop/RunLoop.cpp:
```diff
@@ -235,6 +235,15 @@ void __cdecl main_game_loop_hook() {
 
+typedef void(_cdecl* main_loop_body_t)();
+main_loop_body_t p_main_loop_body_dedi;
+void __cdecl main_loop_body_dedi() 
```
I think it's possible to use `main_game_loop_hook` instead of creating another function. `shouldQuitMainLoop` and `dedicated_server_should_quit_hook` can then be safely removed.

Kantanomo replied:
Yeah should work to just combine the two, and add a check for map downloader and local user stats to only run on client.

Berthalamew replied:
Combined the two main_loop_body functions and removed shouldQuitMainLoop as well as dedicated_server_should_quit_hook

Kantanomo commented on xlive/H2MOD/Variants/Infection/Infection.cpp:
```diff
@@ -259,10 +225,51 @@ void Infection::onGameTick()
+
+	//Replace vehicles with shotgun ammo
+	s_scenario_group_definition* scenario = get_global_scenario();
+	for (DWORD i = 0; i < scenario->netgame_equipment.size; i++)
```
I fail to understand what this change accomplishes it ends up doing the same thing.

Berthalamew replied:
If you're referring to getting the scnr, it does the same thing but instead of creating a for loop and searching for the scnr tag we can just grab the global one

Kantanomo replied:
no the entire thing.

Berthalamew replied:
If the collection ref is null and the classification is a vehicle, vehicles can still spawn even when that ref is null since it'll pull a default from the gametype

Kantanomo replied:
That was intentionally not fixed for vairants that explicitly wanted vehicles.

Berthalamew replied:
Changed

Berthalamew committed:
add requested changes

- make it so gametypes can override vehicles in zombies
- cleanup logic in runloop and have one main_loop_body function for the client and dedi

Kantanomo reviewed (approved):
Changes requested were made, looks good to me just unless Nuke sees something wrong.

except... 
`const kill_volume_enable_or_disable_t func = (enable_volumes ? scenario_kill_trigger_volume_enable : scenario_kill_trigger_volume_disable);`

could just be an if statement and no need for the global typedef.
